datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -750,6 +749,26 @@ impl ParquetSink {
         }
     }
 
+    /// Create writer properties based upon configuration settings,
+    /// including partitioning and the inclusion of arrow schema metadata.
+    fn create_writer_props(&self) -> Result<WriterProperties> {
+        let props = if !self.parquet_options.global.skip_arrow_metadata {
+            let schema = if self.parquet_options.global.allow_single_file_parallelism {
```

Comment:
Can you explain why different schemas are needed for single file parallelism?

Reply:
This is the potential for hive-style partitioning (across multiple files) which impacts the schema per file. I've added a code comment to explain.